topic09/topic-8.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Practice Questions
+author: Ikaro Costa
+---
+
+Practice Questions
+==================
+
+Question 1
+----------
+
+In COMP 1002, you have successfully built your personal webpage. We are now
+going to automate a few of those processes in building (or updating) your personal
+webpage. For this, we are going to write a shell script to automate the conversion
+from markdown to HTML and also to change the permissions of the generated HTML
+file.
+
+1. Let us convert the file from markdown to HTML. Name your script as
+`publish_webpage`. The script should take the name of the markdown file as parameter
+and convert it to HTML using `pandoc`. For example, if your markdown file is
+`index.md`, you would run it as:
+
+    ```bash
+    ./publish_webpage index.md
+    ```
+
+2. Now, we need to change the permissions of the recently generated file
+`index.html`. Recall that the permissions are:
+
+    * `index.html`:
+        * User must have read and write, but no execute permission.
+        * Group must have read, but no write nor execute permissions.
+        * Other must have read, but no write nor execute permissions.
+
+Question 2
+----------
+
+Now, we improve the shell script `publish_webpage` from Question 1. 
+
+### Input Error Prevention
+
+Imagine
+that you shared your script with a friend but he did not check your implementation.
```

Comment:
Please do not use gendered pronouns to describe general people in English. Prefer instead to use "they" or "them". In this case, change this sentence to be

> Imagine that you shared your script with a friend but **they** did not...